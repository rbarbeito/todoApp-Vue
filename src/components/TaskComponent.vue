<template>
  <ul class="px-3 mt-3 lg:h-4/6 lg:overflow-y-auto">
    <li
      class="p-4 my-6 w-full h-34 rounded-lg bg-gray-50 flex flex-col justify-between"
      v-for="task in taskFilter"
      :key="task.id"
    >
      <div class="flex justify-between items-center">
        <div class="flex flex-col">
          <h4 class="font-bold">{{ task.task }}</h4>
          <h6 class="text-gray-400">{{ task.place }}</h6>
        </div>
        <faIcon
          icon="fa-solid fa-trash"
          class="text-red-500 cursor-pointer"
          @click="deleteTask(task.id)"
        />
      </div>
      <hr class="my-2" />
      <div class="flex justify-between items-center">
        <div class="flex flex-row gap-2">
          <h5 class="font-semibold text-gray-500">{{ formatDate(task.date) }}</h5>
          <h5
            class="font-light text-gray-400"
            v-if="!dayjs(task.date).isBetween(dayjs(), dayjs().add(3, 'days'), null, '[]')"
          >
            {{ dayjs().format('HH:mm') }}
          </h5>
        </div>
        <faIcon
          icon="fa-regular fa-circle-check"
          class="cursor-pointer"
          @click="changeStatus(task.id)"
          :class="task.status == 'pending' ? 'text-gray-300' : 'text-green-500'"
        />
      </div>
    </li>
  </ul>
</template>

<script setup lang="ts">
import type { Task } from '@/interfaces/task.interface';
import { computed } from 'vue';

import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';
import LocalizedFormat from 'dayjs/plugin/localizedFormat';
import isBetween from 'dayjs/plugin/isBetween';
import 'dayjs/locale/es';
import { showTask } from '@/composables/showTask';

dayjs.extend(relativeTime);
dayjs.extend(LocalizedFormat);
dayjs.extend(isBetween);
dayjs.locale('es');

const { formatDate } = showTask();

const taskFilter = computed(() => {
  if (props.filtro != 'all') {
    return props.tasks.filter((el) => el.status === props.filtro);
  }

  return props.tasks;
});

interface Props {
  tasks: Task[];
  filtro: string;
}

const props = defineProps<Props>();

const emit = defineEmits<{
  changeStatus: [id: string];
  deleteTask: [id: string];
}>();

const changeStatus = (id: string) => {
  emit('changeStatus', id);
};

const deleteTask = (id: string) => {
  emit('deleteTask', id);
};
</script>
