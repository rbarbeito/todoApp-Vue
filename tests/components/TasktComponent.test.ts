import { describe, expect, test } from "vitest"
import { mount } from '@vue/test-utils'
import TaskComponent from '@/components/TaskComponent.vue'


describe('<TestComponent />', () => {

  test('should match snapshot', () => {

    const wrapper = mount(TaskComponent, {
      props: {
        tasks: [
          {
            id: "jkakjsha-asasasas-asaas",
            place: "home",
            task: "eat",
            date: "2025-02-01 10:00",
            status: 'pending'
          }
        ],
        filtro: 'all',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()

  })

  // test('', () => {


  //   const wrapper = mount(TaskComponent, {
  //     props: {
  //       tasks: [{

  //       }],
  //       filtro: 'all',
  //     },
  //   })


  // })

})
