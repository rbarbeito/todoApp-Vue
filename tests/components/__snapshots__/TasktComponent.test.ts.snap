// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TestComponent /> > should match snapshot 1`] = `
"<ul class="px-3 mt-3 lg:h-4/6 lg:overflow-y-auto">
  <li class="p-4 my-6 w-full h-34 rounded-lg bg-gray-50 flex flex-col justify-between">
    <div class="flex justify-between items-center">
      <div class="flex flex-col">
        <h4 class="font-bold">eat</h4>
        <h6 class="text-gray-400">home</h6>
      </div>
      <faicon icon="fa-solid fa-trash" class="text-red-500 cursor-pointer"></faicon>
    </div>
    <hr class="my-2">
    <div class="flex justify-between items-center">
      <div class="flex flex-row gap-2">
        <h5 class="font-semibold text-gray-500">1 de febrero de 2025</h5>
        <h5 class="font-light text-gray-400">13:41</h5>
      </div>
      <faicon icon="fa-regular fa-circle-check" class="cursor-pointer text-gray-300"></faicon>
    </div>
  </li>
</ul>"
`;
